### PR TITLE
[#479] Added verbose diagnostic output to AJAX timeout exception.

### LINKED ISSUE #479: Provide more verbose output when iWaitForAjaxToFinish() fails.
Currently error messages when an AJAX call fails are quite short:

```
AfterStep # Drupal\DrupalExtension\Context\MinkContext::afterJavascriptStep()
      Unable to complete AJAX request. (RuntimeException)
```

or 

```
And I wait for AJAX to finish                                                                                 # MinkContext::iWaitForAjaxToFinish()
      Unable to complete AJAX request. (RuntimeException)
```

Could we add to that exception message more information about the event that failed?

### PR DESCRIPTION
**Pull request by AlexSkrypnyk**
Closes #479

## Summary

`iWaitForAjaxToFinish()` used to throw a single-line exception when AJAX timed out - just the word "Unable to complete AJAX request." optionally followed by a JSON blob of Behat step metadata. That gave no clue about *why* the browser was still busy. This PR captures live browser state (current URL, `jQuery.active` count, animated-element count, and each active `Drupal.ajax.instances` entry with its selector, event, and URL) at the moment of timeout and includes it in a structured, human-readable multi-line message. A try/catch guards the extra JS evaluation so the step still fails cleanly when the driver is unavailable or the browser has died.

The Drupal 7 code path (the `for...in Drupal.ajax` branch) has been removed from both the existing wait condition and the new diagnostic capture - the project requires Drupal 8+ only.

## Changes

**`src/Drupal/DrupalExtension/Context/Traits/AjaxTrait.php`**
- Removed dead Drupal 7 `for...in Drupal.ajax` loop from the wait-condition JS.
- Added `captureAjaxDiagnostics()` - runs a second JS snippet on the failure path to collect URL, `jQuery.active`, animated-element count, and active `Drupal.ajax.instances` entries. Returns `NULL` if the driver throws or returns malformed output.
- Added `formatTimeoutMessage()` - a pure static helper that assembles the multi-line exception message from the timeout value, diagnostics array, and optional step-scope data. Pure so it can be unit tested without a real browser.
- Added `formatScalar()` - renders a diagnostic scalar value, mapping `NULL` to `'unavailable'`.
- The existing step-scope data (`name`, `feature`, `step`, `suite`) is preserved but renamed/restructured into the new message format under a `Hook:` heading.

**`tests/phpunit/src/Context/Traits/AjaxTraitTest.php`** (new)
- 12 data-provider test cases for `formatTimeoutMessage()` via a `TestableAjaxConsumer` shim (same pattern as `DeprecationTraitTest`).
- Cases cover: timeout-only, event-only, full diagnostics with one instance, diagnostics + event combined, `jquery_active` 0 vs `NULL`, multiple instances, instance with no known fields, malformed instance entry, missing URL key, non-string URL, non-array `drupal_ajax_instances`.

## Before / After

```
BEFORE (single line, no browser context):
+-----------------------------------------------------------------+
| RuntimeException:                                               |
| Unable to complete AJAX request. {"name":"step.before",        |
| "feature":"My Feature","step":"I click button","suite":"dflt"} |
+-----------------------------------------------------------------+

AFTER (multi-line, full browser context):
+-----------------------------------------------------------------+
| RuntimeException:                                               |
| Unable to complete AJAX request after 5 second(s).             |
|                                                                 |
| URL: http://example.com/node/add                               |
| jQuery active requests: 2                                       |
| jQuery animated elements: 0                                     |
| Drupal AJAX instances active: 1                                 |
|   - selector: '#edit-submit', event: 'mousedown',              |
|     url: '/system/ajax', index: '0'                            |
|                                                                 |
| Hook: step.before                                               |
| Feature: My Feature                                             |
| Step: I click button                                            |
| Suite: default                                                  |
+-----------------------------------------------------------------+
```

## Summary by CodeRabbit

* **Bug Fixes**
  * Enhanced AJAX timeout error messages with comprehensive diagnostic information including current URL, jQuery request counts, and active operations, enabling faster troubleshooting of timeout issues.

* **Tests**
  * Added comprehensive test coverage for AJAX timeout message formatting, validating correct behavior across diverse scenarios including edge cases with missing or malformed diagnostic data.

### CODE REVIEW

File: src/Drupal/DrupalExtension/Context/Traits/AjaxTrait.php [hunk 1]
```diff
@@ -103,21 +100,151 @@ function isAjaxing(instance) {
...
+
+      foreach ($instances as $instance) {
+        if (!is_array($instance)) {
+          continue;
+        }
+
+        $parts = [];
+        foreach (['selector', 'event', 'url', 'index'] as $field) {
+          if (isset($instance[$field])) {
+            $parts[] = sprintf("%s: '%s'", $field, $instance[$field]);
```
[the remark this answers is not in the document]
Comment (AlexSkrypnyk) [reply src/Drupal/DrupalExtension/Context/Traits/AjaxTrait.php]: Fixed in 322bda5. Added a `redactUrl()` helper that strips query and fragment from both the page URL and each Drupal AJAX instance URL before formatting. New test cases cover URL redaction; existing fixtures (which used path-only URLs) still pass.
